mpt/impl/imap-mailbox/rabbitmq/src/test/java/org/apache/james/mpt/imapmailbox/rabbitmq/host/RabbitMQEventBusHostSystem.java

```diff
@@ -0,0 +1,197 @@
+import org.apache.james.mailbox.events.RabbitMQEventBus;
+import org.apache.james.mailbox.events.RetryBackoffConfiguration;
+import org.apache.james.mailbox.events.RoutingKeyConverter;
+import org.apache.james.mailbox.inmemory.InMemoryId;
+import org.apache.james.mailbox.inmemory.InMemoryMailboxManager;
+import org.apache.james.mailbox.inmemory.InMemoryMailboxSessionMapperFactory;
+import org.apache.james.mailbox.inmemory.InMemoryMessageId;
+import org.apache.james.mailbox.inmemory.quota.InMemoryCurrentQuotaManager;
+import org.apache.james.mailbox.inmemory.quota.InMemoryPerUserMaxQuotaManager;
+import org.apache.james.mailbox.store.JVMMailboxPathLocker;
+import org.apache.james.mailbox.store.MailboxSessionMapperFactory;
+import org.apache.james.mailbox.store.SessionProvider;
+import org.apache.james.mailbox.store.StoreMailboxAnnotationManager;
+import org.apache.james.mailbox.store.StoreMailboxManager;
+import org.apache.james.mailbox.store.StoreRightManager;
+import org.apache.james.mailbox.store.StoreSubscriptionManager;
+import org.apache.james.mailbox.store.extractor.DefaultTextExtractor;
+import org.apache.james.mailbox.store.mail.model.impl.MessageParser;
+import org.apache.james.mailbox.store.quota.CurrentQuotaCalculator;
+import org.apache.james.mailbox.store.quota.DefaultUserQuotaRootResolver;
+import org.apache.james.mailbox.store.quota.ListeningCurrentQuotaUpdater;
+import org.apache.james.mailbox.store.quota.QuotaComponents;
+import org.apache.james.mailbox.store.quota.StoreQuotaManager;
+import org.apache.james.mailbox.store.search.MessageSearchIndex;
+import org.apache.james.mailbox.store.search.SimpleMessageSearchIndex;
+import org.apache.james.metrics.logger.DefaultMetricFactory;
+import org.apache.james.mpt.api.ImapFeatures;
+import org.apache.james.mpt.api.ImapFeatures.Feature;
+import org.apache.james.mpt.host.JamesImapHostSystem;
+import org.apache.james.util.concurrent.NamedThreadFactory;
+
+import com.google.common.collect.ImmutableSet;
+import com.nurkiewicz.asyncretry.AsyncRetryExecutor;
+
+public class RabbitMQEventBusHostSystem extends JamesImapHostSystem {
+
+    private static final ImapFeatures SUPPORTED_FEATURES = ImapFeatures.of(Feature.NAMESPACE_SUPPORT,
+        Feature.MOVE_SUPPORT,
+        Feature.USER_FLAGS_SUPPORT,
+        Feature.QUOTA_SUPPORT,
+        Feature.ANNOTATION_SUPPORT,
+        Feature.MOD_SEQ_SEARCH);
+
+
+    private static final int THREE_RETRIES = 3;
+    private static final int ONE_HUNDRED_MILLISECONDS = 100;
+
+    private final DockerRabbitMQ dockerRabbitMQ;
+    private StoreMailboxManager mailboxManager;
+    private InMemoryPerUserMaxQuotaManager maxQuotaManager;
+    private RabbitMQConnectionFactory rabbitConnectionFactory;
+    private RabbitMQEventBus eventBus;
+
+    public RabbitMQEventBusHostSystem(DockerRabbitMQ dockerRabbitMQ) {
+        this.dockerRabbitMQ = dockerRabbitMQ;
+    }
+
+    @Override
+    public void beforeTest() throws Exception {
```

Comment:
is part of the setup is used some where else before, I think so because the setup step could be almost the same while the only difference is you created RabbitMQ eventbus instead of InVmEventBus.
Could the same part being extracted to a method or an extension?

Reply:
> Could the same part being extracted to a method or an extension?

I give up trying to generify the InMemory integration resources. Overriding 'just' the event bus is actually not that easy. And introducing a new method per-component do not seems good either.

Hence I decided to not perform such a task and explicitly write the all wiring required here.